src/Filesystem/Dir.php

```diff
@@ -139,23 +139,31 @@ public static function files(
 
 	/**
 	 * Read the directory and all subdirectories
+	 *
+	 * @todo Remove support for `$ignore = null` in a major release
+	 * @param array|null $ignore Passing null is deprecated.
 	 */
 	public static function index(
 		string $dir,
 		bool $recursive = false,
-		array|null $ignore = [],
+		array|bool|null $ignore = [],
```

Comment:
I think PHP actually allows to type hint for `array|false|null` - otherwise we'd also need to handle `true`.